camel/o1datagen/o1datagen.py

```diff
@@ -0,0 +1,335 @@
+# ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
+
+
+import json
+from collections import defaultdict
+from datetime import datetime
+from typing import Dict
+
+from camel.agents import ChatAgent
+from camel.logger import enable_logging, get_logger
+
+# Enable logging if needed
+enable_logging()
+
+# Get a logger for this module
+logger = get_logger('o1datagen')
+
+
+class O1DataGene:
```

Comment:
Should this be called O1DataGenerator? Or is DataGene the name of something that this is referencing?

Reply:
yeah ,the examples and cookbook use this  class to instantiation

Reply:
Let's use `O1DataGenerator` instead of using the abbreviation in the class name